clients/redis/command.go

```diff
@@ -0,0 +1,308 @@
+	requestEnvelope := types.NewMessageEnvelopeForRequest(nil, queryParams)
+	requestTopic := strings.Join([]string{c.topics[QueryRequestTopic], common.All}, "/")
+	err := c.messageBus.Publish(requestEnvelope, requestTopic)
+	if err != nil {
+		return responses.MultiDeviceCoreCommandsResponse{}, err
+	}
+
+	select {
+	case <-ctx.Done():
+		return responses.MultiDeviceCoreCommandsResponse{}, nil
+	case <-time.After(c.timeout):
+		return responses.MultiDeviceCoreCommandsResponse{}, errors.New("timed out waiting response")
+	case err = <-c.queryErrors:
+		return responses.MultiDeviceCoreCommandsResponse{}, err
+	case responseEnvelope := <-c.queryMessages:
+		if responseEnvelope.ErrorCode == 1 {
+			return responses.MultiDeviceCoreCommandsResponse{}, errors.New(string(responseEnvelope.Payload))
+		}
+
+		var res responses.MultiDeviceCoreCommandsResponse
+		err = json.Unmarshal(responseEnvelope.Payload, &res)
+		if err != nil {
+			return responses.MultiDeviceCoreCommandsResponse{}, err
+		}
+
+		return res, nil
+	}
+}
+
+func (c *CommandClient) DeviceCoreCommandsByDeviceName(ctx context.Context, name string) (responses.DeviceCoreCommandResponse, error) {
+	if c.queryMessages == nil {
+		return responses.DeviceCoreCommandResponse{}, errors.New("commandquery request/response topics not provided")
+	}
+
+	requestEnvelope := types.NewMessageEnvelopeForRequest(nil, nil)
+	requestTopic := strings.Join([]string{c.topics[QueryRequestTopic], name}, "/")
+	err := c.messageBus.Publish(requestEnvelope, requestTopic)
+	if err != nil {
+		return responses.DeviceCoreCommandResponse{}, err
+	}
+
+	select {
+	case <-ctx.Done():
+		return responses.DeviceCoreCommandResponse{}, nil
+	case <-time.After(c.timeout):
+		return responses.DeviceCoreCommandResponse{}, errors.New("timed out waiting response")
+	case err = <-c.queryErrors:
+		return responses.DeviceCoreCommandResponse{}, err
+	case responseEnvelope := <-c.queryMessages:
+		if responseEnvelope.ErrorCode == 1 {
+			return responses.DeviceCoreCommandResponse{}, errors.New(string(responseEnvelope.Payload))
+		}
+
+		var res responses.DeviceCoreCommandResponse
+		err = json.Unmarshal(responseEnvelope.Payload, &res)
+		if err != nil {
+			return responses.DeviceCoreCommandResponse{}, err
+		}
+
+		return res, nil
+	}
+}
+
+func (c *CommandClient) IssueGetCommandByName(ctx context.Context, deviceName string, commandName string, dsPushEvent bool, dsReturnEvent bool) (responses.EventResponse, error) {
+	if c.commandMessages == nil {
+		return responses.EventResponse{}, errors.New("command request/response topics not provided")
+	}
+
+	pushEvent, returnEvent := "no", "no"
+	if dsPushEvent {
+		pushEvent = "yes"
+	}
+	if dsReturnEvent {
+		returnEvent = "yes"
+	}
+	queryParams := map[string]string{common.PushEvent: pushEvent, common.ReturnEvent: returnEvent}
+	requestEnvelope := types.NewMessageEnvelopeForRequest(nil, queryParams)
```

Comment:
Can this just call `IssueGetCommandByNameWithQueryParams` here now that we have with `dsPushEvent` & `dsReturnEvent` added as the queryParameters .